#164069179 reset password link

Peace-Apple opened the pull request:
## What does this PR do?

This PR implements the feature of a user being provided with a reset password link

## Description of the task to be completed

- Enable users to provide their email to reset their password
- Provide the users with a reset password link
- Enable users to change their password

## How should this be manually tested?

- Clone this repository
- cd into the directory Ah-aquaman-frontend
- Check out the branch `ft-reset-password-164069179`
- Run npm install to install all dependencies
- Run npm start to get the server running
- You will view the reset password page at `localhost/reset_password`
- Once you click on the link sent to your email, you are redirected to the page where you actually change your password `localhost/reset_password/token`

## What are the relevant pivotal tracker stories?

[#164069179](https://www.pivotaltracker.com/story/show/164069179)

## Screenshots

Reset password page

![image](https://user-images.githubusercontent.com/42435346/56277650-6b0d1780-610d-11e9-8595-5940274c8f78.png)


Change password page

![image](https://user-images.githubusercontent.com/42435346/55956837-90ada300-5c6d-11e9-843a-9d46425beac9.png)

joelethan commented on src/__tests__/actions/resetAction/__test__/changePasswordActions.test.js:
```diff
@@ -0,0 +1,73 @@
+const mockStore = configureMockStore(middlewares);
+const { BASE_URL } = process.env;
+
+
```
@Peace-Apple Please clear these blank-lines for more readability.

Peace-Apple replied:
Thanks Joel let me fix them

Peace-Apple replied:
I have fixed them now @joelethan

joelethan commented on src/__tests__/actions/resetAction/__test__/changePasswordActions.test.js:
```diff
@@ -0,0 +1,73 @@
+      headers: {
+           "content-type": "application/json", 
+           'Authorization': `Bearer ${token}`},
+      
```
I think this blankline needs fixing too.

Peace-Apple replied:
Thanks let me fix them

Peace-Apple replied:
I have fixed them now

joelethan reviewed (changes requested):
There are some redundant blank-lines that need to be refactored.

joelethan reviewed (commented):
@Peace-Apple  Am seeing lint errors when I run "npm run lint". I think they need to be worked on.

Peace-Apple commented:
> @Peace-Apple Am seeing lint errors when I run "npm run lint". I think they need to be worked on.

Thanks Joel, Am going to fix them

Peace-Apple committed:
feat(passwordReset): users should be presented with a reset password link

- enable users to enter their email to receive password reset link
- enable users to change their passwords

[Starts #164069179]

joelethan commented:
@Peace-Apple Please check the commit message, it needs to match Andela standards.

joelethan commented on src/__tests__/views/resetView/__test__/resetPasswordView.test.js:
```diff
@@ -0,0 +1,117 @@
+    instance.componentWillReceiveProps({
+      preventDefault: jest.fn(),
+    });
+    // expect(props.componentWillReceiveProps).toHaveBeenCalled();
```
@Peace-Apple please remove this commented out code

Peace-Apple replied:
Thanks, I have worked on that

BagzieGracious commented on src/actions/resetAction/changePasswordActions.js:
```diff
@@ -0,0 +1,30 @@
+})
+  .then(res => res.json())
+  .then((data) => { 
+    console.log(data);
```
Thank you for the good work peace, but please remove this console.log

Peace-Apple replied:
Thanks, @BagzieGracious I have removed that

BagzieGracious commented on src/reducers/resetReducer/resetPasswordReducer.js:
```diff
@@ -0,0 +1,23 @@
+import { RESET_PASSWORD_FAIL, RESET_PASSWORD_SUCCESS } from "../../actions/types";
+
+const initialState = {};
```
Thank you for the good work @Peace-Apple, but please can you remove this constant since it's empty and I think it's better if you use `state = {}`

Peace-Apple replied:
Okay @BagzieGracious I have changed that

CryceTruly reviewed (changes requested):
Good work @Peace-Apple, You only have to add a progress indicator when asynchronous operations are going on.

Peace-Apple commented:
> Good work @Peace-Apple, You only have to add a progress indicator when asynchronous operations are going on.

Could you please elaborate more on this, I do not quite get what you mean exactly

Peace-Apple commented:
> Good work @Peace-Apple, You only have to add a progress indicator when asynchronous operations are going on.

About this, I have now added a loader @CryceTruly

BagzieGracious reviewed (approved):
Thank you for the good work @Peace-Apple, you have worked on feedback

AmosWels commented on src/views/resetView/resetPasswordView.js:
```diff
@@ -0,0 +1,133 @@
+              pauseOnHover: true,
+            });
+          }
+
```
remove this unnecessary blank line in the same block

Peace-Apple replied:
I have worked on this, thanks

AmosWels commented on src/__tests__/views/resetView/__test__/resetPasswordView.test.js:
```diff
@@ -0,0 +1,116 @@
+    });
+  });
+
+  it("should handle onChange with correct email input", () => {
```
i find that this test is repeated. You are testing the `onChange` function. Also add an assertion to it.

Peace-Apple replied:
I have worked on this and added the assertions, and about the repetition, I have changed them, the first case covers when a correct email is entered and then the second case looks at when a wrong email is entered so it returns an error, thanks

AmosWels commented on src/__tests__/views/resetView/__test__/resetPasswordView.test.js:
```diff
@@ -0,0 +1,116 @@
+    instance.onChange({ target: { name: "email", value: "apple@acio.com" } });
+  });
+
+  it("should handle onChange with wrong email input", () => {
```
it's repeated here

Peace-Apple replied:
I have now changed it

AmosWels commented on src/__tests__/views/resetView/__test__/resetPasswordView.test.js:
```diff
@@ -0,0 +1,116 @@
+    instance.onChange({ target: { name: "email", value: "appleacio.com" } });
+  });
+    
+  it("should call component will receive props", () => {
```
add an assertion to this test. `expect` some action after

AmosWels reviewed (changes requested):
Goodwork @Peace-Apple , improve the formatting on your PR such that details like branch name are well formatted.

Also run `npm run lint` and check the console. Remove all the Unused state fields in the `editProfile.js` page.

Peace-Apple commented:
> Goodwork @Peace-Apple , improve the formatting on your PR such that details like branch name are well formatted.
> 
> Also run `npm run lint` and check the console. Remove all the Unused state fields in the `editProfile.js` page.

> @CryceTruly uses these fields but he will destructure them in his next PR
